### ISSUE TITLE
IMDB datset

### ISSUE THREAD

**Issue by sohamranade** [microsoft/UST#6]
I am trying to replicate the model's performance on IMDB dataset but I am not able to do so? 
Does this method essentially work on only binary classification? I have tried to make it work with multiple classes but the model doesn't run as I get gradient doesn't exist error?
Can the details regarding the parameters used and the way dataset was created be shared so that we can replicate and test the model?

**Comment by addiu** [comment]
Hi, I observe a similar issue as yours. I tried the code on SST-2 and AG-News, I cannot reproduce the results, either. Have you tried the code on these two datasets?
I see the model fine-tuned on the 30 labeled data points per class already performs well, and the UST does not push the performance much further, <=1% improvement. Did you observe the same?

**Comment by sohamranade** [comment]
Hey, I don't have sufficient compute power to keep it running for the epochs mentioned in the paper to fully replicate the findings. I did train it on these datasets and I got similar results to you, With IMDB however I was getting really bad accuracy in the first place, so wanted to know if there was some other issue with the approach.
I have found the following things which helped:
The labels can be multiclass labels but should start from 0 to n-1 for n classes. 
Sequence length is an important parameter.
Please let me know if you could replicate the model fully on your end.

**Comment by addiu** [comment]
Hi, thanks a lot for the quick reply! I will also try it on IMDB and let you know the results from my side.
Some notes regarding your comments:
> The labels can be multiclass labels but should start from 0 to n-1 for n classes.

yes. I think the code also did it in `run_ust.py`.  it deducts 1 for every class to have a range of [0, n-1]

>Sequence length is an important parameter. 

I agree. I tried out the sequence length reported in the appendix of the paper. I still cannot reproduce the results.

I am still trying to reproduce the results and will update my progress here.

**Comment by subhomj** [comment]
For improved UST performance, refer to the ReadME section on "Continued Pre-training on Task-specific Unlabeled Data". You can pull in the pre-trained checkpoint for IMDB from the link mentioned there to initialize UST.

For exact replication, use the original code used in the paper (prior to HuggingFace integration) arxived in NeurIPS website https://proceedings.neurips.cc/paper/2020/file/f23d125da1e29e34c552f448610ff25f-Supplemental.zip. The fine-tuning difference could be due to the underlying DL package.

**Comment by sohamranade** [comment]
@subhomj  Thanks for the inputs
